src/dev.ts

```diff
@@ -0,0 +1,279 @@
+
+      console.log(colors.green('✅ Development server started'))
+      console.log(colors.gray(`   Dashboard: http://localhost:${this.options.dashboardPort || 3000}`))
+      console.log(colors.gray(`   Application: http://localhost:${this.options.port || 8000}`))
+    } catch (error) {
+      console.error(colors.red('❌ Failed to start development server:'), error)
+      throw error
+    }
+  }
+
+  private printWelcomeBanner () {
+    console.log(colors.cyan('╔══════════════════════════════════════════════════════════════╗'))
+    console.log(colors.cyan('║') + colors.bold(colors.white('               🧪 CHEL LIVE DEVELOPMENT                     ')) + colors.cyan('║'))
+    console.log(colors.cyan('║') + colors.white('           Live-testing with Hot Reload & Interaction        ') + colors.cyan('║'))
+    console.log(colors.cyan('╚══════════════════════════════════════════════════════════════╝'))
+    console.log()
+    console.log(colors.gray('Similar to grunt dev - provides live-testing environment'))
+    console.log(colors.gray('with automatic contract redeployment and interactive testing.'))
+    console.log()
+  }
+
+  private async startContractWatching () {
+    console.log(colors.blue('👀 Setting up contract file watching...'))
+    const contractsDir = join(this.projectRoot, 'contracts')
+
+    if (!existsSync(contractsDir)) {
+      console.log(colors.yellow('⚠️  No contracts directory found, creating one...'))
+      await mkdir(contractsDir, { recursive: true })
+      return
+    }
+
+    try {
+      // Watch for changes in contract files
+      const watcher = watch(contractsDir, { recursive: true }, (eventType, filename) => {
+        if (filename && (filename.endsWith('.js') || filename.endsWith('.json'))) {
+          this.handleContractChange(eventType, filename)
+        }
+      })
+
+      this.watchers.push(watcher)
+
+      // Count initial contracts
+      const entries = await readdir(contractsDir, { recursive: true })
+      const contractFiles = (entries as unknown as string[]).filter((f) =>
+        typeof f === 'string' && f.endsWith('.js')
+      )
+      this.metrics.contractsWatched = contractFiles.length
+
+      console.log(colors.green(`✅ Watching ${contractFiles.length} contract files for changes`))
+    } catch (error) {
+      console.error(colors.red('❌ Error setting up file watching:'), error)
+    }
+  }
+
+  private handleContractChange (eventType: string, filename: string) {
+    const changeEvent: ContractChangeEvent = {
+      file: filename,
+      type: eventType === 'rename' ? 'added' : 'changed',
+      timestamp: new Date()
+    }
+
+    this.contractChanges.push(changeEvent)
+    this.metrics.changesDetected++
+
+    console.log(colors.yellow(`📝 Contract ${eventType}: ${filename}`))
+
+    if (this.options['hot-reload'] !== false) {
+      this.triggerHotReload(filename)
+    }
+  }
+
+  private triggerHotReload (filename: string) {
```

Comment:
This function is apparently doing nothing and it needs to reload contracts as per your comment below. Note that this is wrong: 'Re-pin the changed contract'. Development contracts are not and should not be pinned.